### ISSUE TITLE
bm_*  tests are not being run on bazel/foundry

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#16850]
The `bm_*` tests  are running on kokoro but they are not being run on foundry.

List of test cases being run on kokoro:
https://source.cloud.google.com/results/invocations/e2ed4007-17f2-471c-8bd7-eb4b06d7b94c/targets/github%2Fgrpc%2Fcpp_linux_dbg_native/tests;query=bm_
```
bm_arena
bm_call_create
bm_channel
bm_chttp2_hpack
bm_chttp2_transport
bm_closure
bm_cq
bm_cq_multiple_threads
bm_error
bm_fullstack_streaming_ping_pong
bm_fullstack_streaming_pump
bm_fullstack_trickle
bm_fullstack_unary_ping_pong
bm_metadata
bm_pollset
```

The BUILD file with cc_binary files exists, but we are still not running those on Foundry (there's probably a set of rules missing)
https://github.com/grpc/grpc/blob/master/test/cpp/microbenchmarks/BUILD

- do we want to run those on Foundry? Running tests is useful, but these ones are clearly benchmarks masked as unit tests. It would be good to have a smoketest run that would finish quickly (so that we are not wasting resources) but still make sure the microbenchmarks work.
- if we do want to run these, we will probably need to generate rules to make the benchmarks runnable on Foundry.

**Comment by jtattermusch** [comment]
@ncteisen sg, please add a bazel rule for those tests then.

**Comment by ncteisen** [comment]
Just changed the binary rules to test rules. Except for trickle, we don't want that one running as a unit test because it takes too long, and has its own dedicated job

**Comment by ncteisen** [comment]
#16882

**Comment by jtattermusch** [comment]
Fixed by #16882
